### PR TITLE
Add TransactionCard, TransactionListPanel component, and split panel layout

### LINKED ISSUE #94: Add new UI component for transactions

### PR DESCRIPTION
**Pull request by GavinSin**
- [X]  This pull request is complete
- [ ]  This pull request is a work in progress

### Description
- update `Logic`, `LogicManager` for dependencies needed for new ui
- add window listener that switch between split screen and one big screen
  - if screen is less than `800px`, then there is only a big screen
  - the big screen will display either person list or transaction list based on last command
  - else, it will show split view where the left side is original ui and the right side is transaction list
- fix entry point in build.gradle

close #94

### CODE REVIEW

**Comment by GavinSin** [comment]
Hi @AY2425S1-CS2103-F13-1/developers , is there a need for me to write test cases for `TransactionCard`, `TransactionListPanel` component? A possible todo for myself is a test case on `adjustPaneVisibility` in `MainWindow.java`.

**Comment by SeeYangZhi** [comment]
> Hi @AY2425S1-CS2103-F13-1/developers, for GUI testing, thinking of using [TestFx](https://github.com/TestFX/TestFX) library, then test for each command, the output in `resultDisplayPlaceholder`, `personListView`, `transactionListView` are displayed correctly. In doing so, the testing should go through all the required components and ticked off code coverage issues. So no need _'unit testing'_ each individual components like `TransactionCard`, `TransactionListPanel` that is not helpful.
> 
> TODO (after completion of `AddTxn` and `ListTxn` command):
> 
> * [ ]  GUI testing for add txn then output correctly for both big screen and split screen
> * [ ]  a test case on `adjustPaneVisibility` in `MainWindow.java`

[TestFx](https://github.com/TestFX/TestFX) is in line with what the CS2103 textbook recommends. Unless you want to learn selenium. Else we can go ahead with testFX

**Comment by SeeYangZhi** [comment]
@GavinSin You may explore creating Draft PRs if it is still work in progress

**Comment by seanlim** [comment]
Adding `do not merge` label as this is still WIP

File: src/test/java/spleetwaise/address/testutil/Assert.java [hunk 1]
```diff
@@ -1,34 +1,85 @@
...
+     * Asserts that the actual card displays the expected person's details correctly.
+     *
+     * @param expectedPerson The person whose details are expected to be displayed.
+     * @param actualCard     The actual card handle that displays the person's details.
+     */
+    public static void assertCardDisplaysPerson(Person expectedPerson, PersonCardHandle actualCard) {
+        assertEquals(expectedPerson.getName().fullName, actualCard.getName());
+        assertEquals(expectedPerson.getPhone().value, actualCard.getPhone());
+        assertEquals(expectedPerson.getEmail().value, actualCard.getEmail());
+        assertEquals(expectedPerson.getAddress().value, actualCard.getAddress());
```
Comment (rollingpencil) [inline src/test/java/spleetwaise/address/testutil/Assert.java:56]: There is a missing assertion for the remark field.

**Review by rollingpencil — DISMISSED** [review]
Looks good to me overall! Just a few nits that you might want to address before merging.

File: src/main/java/spleetwaise/address/ui/MainWindow.java [hunk 2]
```diff
@@ -100,13 +127,71 @@ private void setAccelerator(MenuItem menuItem, KeyCombination keyCombination) {
...
+    }
+
+    /**
+     * Checks if the command is related to transactions.
+     *
+     * @param command The command to check.
+     * @return True if the command is 'addTxn' or 'listTxn', false otherwise.
+     */
+    private boolean isTransactionCommand(String command) {
+        return command.equals("addTxn") || command.equals("listTxn");
```
[the remark this answers is not in the document]
Comment (GavinSin) [reply src/main/java/spleetwaise/address/ui/MainWindow.java:149]: you are right, its better to have isTransactionCommand in LogicManager instead.

**Review by rollingpencil — APPROVED** [review]
The updated changes looks good to me!

**Pull request merged**